browser/farbling/brave_navigator_keyboard_api_browsertest.cc

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/containers/contains.h"
+#include "base/path_service.h"
+#include "brave/browser/brave_content_browser_client.h"
+#include "brave/common/brave_paths.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/common/chrome_content_client.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/network_session_configurator/common/network_switches.h"
+#include "components/permissions/permission_request.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "net/dns/mock_host_resolver.h"
+#include "url/gurl.h"
+
+using brave_shields::ControlType;
+
+const char kGetLayoutMapScript[] =
+    "(async function testKeyboardAPI() {"
+    "  keyboardLayoutMap = await navigator.keyboard.getLayoutMap();"
+    "  return keyboardLayoutMap.get('KeyA');"
+    "})()";
+
+class BraveNavigatorKeyboardAPIBrowserTest : public InProcessBrowserTest {
+ public:
+  BraveNavigatorKeyboardAPIBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    https_server_.SetSSLConfig(net::EmbeddedTestServer::CERT_OK);
+    https_server_.ServeFilesFromDirectory(test_data_dir);
+    EXPECT_TRUE(https_server_.Start());
+    top_level_page_url_ = https_server_.GetURL("a.com", "/");
+    test_url_ = https_server_.GetURL("a.com", "/simple.html");
+  }
+
+  BraveNavigatorKeyboardAPIBrowserTest(
+      const BraveNavigatorKeyboardAPIBrowserTest&) = delete;
+  BraveNavigatorKeyboardAPIBrowserTest& operator=(
+      const BraveNavigatorKeyboardAPIBrowserTest&) = delete;
+
+  ~BraveNavigatorKeyboardAPIBrowserTest() override {}
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    // HTTPS server only serves a valid cert for localhost, so this is needed
+    // to load pages from other hosts without an error
+    command_line->AppendSwitch(switches::kIgnoreCertificateErrors);
```

Comment:
better avoid this (as per annoucement on slack)

Reply:
Fixed in 0dfc2a865d79406aacbb4dbf353e0b91f6d3846b